### PR TITLE
Rke2 ci testing

### PR DESCRIPTION
**Pull request by jacobbmay**
PR for #13 

Changes:
* Added workflow and terraform for testing creating an RKE2 cluster with AMIs built from this repo
* Packer now saves a manifest that can be used in CI to grab the AMI ID for creating EC2s and cleaning up AMIs and snapshots after CI tests
* Added slash commands for dispatching test install jobs so they can be run when needed in a PR, but don't run on every commit

### CODE REVIEW

File: .github/workflows/test-rke2-install-generic.yaml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          terraform init -force-copy \
+            -backend-config="bucket=uds-ci-state-bucket" \
+            -backend-config="key=tfstate/ci/install/${SHA:0:7}-rke2.tfstate" \
+            -backend-config="region=us-west-2" \
+            -backend-config="dynamodb_table=uds-ci-state-dynamodb"
+          terraform plan -var="ami_id=$TEST_AMI_ID" -out=rke2-plan
+          sleep 15 # time to review
+          terraform apply -var="ami_id=$TEST_AMI_ID" -auto-approve rke2-plan
+          kubectl apply -k "github.com/kubernetes-sigs/aws-ebs-csi-driver/deploy/kubernetes/overlays/stable/?ref=release-1.21"
+          kubectl apply -f storage-class.yaml
```
Comment (mjnagel) [inline .github/workflows/test-rke2-install-generic.yaml:118]: Imo for CI we can drop the plan and sleep, it doesn't seem likely that we'd be watching a job and cancel out in time. (I'll probably update the DUBBD workflow similarly).

Also not sure we need to apply the driver + storage class, although it is a nice test that the cluster is working. Maybe instead we can do something simpler/quicker to validate the nodes are accessible? `k get nodes` or similar is the extent I was thinking.

File: .github/actions/comment/action.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Update Slash Command Dispatch Comment
```
Comment (mjnagel) [inline .github/actions/comment/action.yaml:1]: Want to discuss whether to use slash commands...can have that conversation offline but wanted to open a thread here to track decision/pros/cons.

File: .github/workflows/test-rke2-install-generic.yaml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      - name: Grab AMI ID
+        run: echo "TEST_AMI_ID=$(jq -r '.builds[-1].artifact_id' manifest.json | cut -d ":" -f2)" >> $GITHUB_ENV
+
+      - name: Create RKE2 cluster
+        env:
+          SHA: ${{ github.sha }}
+        run: |
+          terraform init -force-copy \
+            -backend-config="bucket=uds-ci-state-bucket" \
+            -backend-config="key=tfstate/ci/install/${SHA:0:7}-rke2.tfstate" \
```
Comment (mjnagel) [inline .github/workflows/test-rke2-install-generic.yaml:111]: Small comment - might be nice to name state to differentiate from the DUBBD deployments. SHA will differentiate already but maybe a `packer` or similar to clearly label.

File: .github/workflows/rke2/main.tf [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+disable:
+  - rke2-ingress-nginx
+  - rke2-metrics-server
+EOF
+  controlplane_internal       = var.controlplane_internal
+  associate_public_ip_address = var.associate_public_ip_address
+  wait_for_capacity_timeout   = "20m"
+}
+
+module "agents" {
```
[the remark this answers is not in the document]
Comment (jacobbmay) [reply .github/workflows/rke2/main.tf:39]: I think that is fine. I just copied the workfllow and terraform from dubbd and just removed the dubbd part.

File: .github/workflows/test-rke2-install-generic.yaml (hunk 1 above)
Comment (jacobbmay) [reply .github/workflows/test-rke2-install-generic.yaml:118]: Yeah I guess if this is just testing that the module succeeds standing up a cluster we don't really need to save the plan or review it.

I almost removed those apply steps but just left them to show being able interact with the running cluster. Could definitely swap them out for something simpler and faster though.

File: .github/actions/comment/action.yaml (hunk 2 above)
Comment (mjnagel) [reply .github/actions/comment/action.yaml:1]: Based on convo offline I think we will proceed with the slash commands as is for now - can reevaluate in the future based on what seems common across other repos.

File: Makefile [hunk 5]
```diff
@@ -27,6 +31,44 @@ test-ami-ubuntu: fmt-ami validate-ami-ubuntu build-ami-ubuntu ## fmt, validate,
...
+	TEST_AMI_ID=$$(jq -r '.builds[-1].artifact_id' $(AWS_DIR)/manifest.json | cut -d ":" -f2); \
+	cd $(TEST_TF_DIR); \
+	terraform init -force-copy \
+		-backend-config="bucket=uds-ci-state-bucket" \
+		-backend-config="key=tfstate/ci/install/$${SHA:0:7}-packer-rhel-rke2.tfstate" \
+		-backend-config="region=us-west-2" \
+		-backend-config="dynamodb_table=uds-ci-state-dynamodb"; \
+	terraform apply -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
+	kubectl get nodes; \
+	terraform destroy -auto-approve; \
```
Comment (mjnagel) [inline Makefile:64]:
```suggestion
	terraform destroy -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
```
I think maybe? Looks like the teardown failed here.

File: Makefile [hunk 6]
```diff
@@ -27,6 +31,44 @@ test-ami-ubuntu: fmt-ami validate-ami-ubuntu build-ami-ubuntu ## fmt, validate,
...
+	TEST_AMI_ID=$$(jq -r '.builds[-1].artifact_id' $(AWS_DIR)/manifest.json | cut -d ":" -f2); \
+	cd $(TEST_TF_DIR); \
+	terraform init -force-copy \
+		-backend-config="bucket=uds-ci-state-bucket" \
+		-backend-config="key=tfstate/ci/install/$${SHA:0:7}-packer-ubuntu-rke2.tfstate" \
+		-backend-config="region=us-west-2" \
+		-backend-config="dynamodb_table=uds-ci-state-dynamodb"; \
+	terraform apply -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
+	kubectl get nodes; \
+	terraform destroy -auto-approve; \
```
Comment (mjnagel) [inline Makefile:45]:
```suggestion
	terraform destroy -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
```

File: Makefile [hunk 7]
```diff
@@ -27,6 +31,44 @@ test-ami-ubuntu: fmt-ami validate-ami-ubuntu build-ami-ubuntu ## fmt, validate,
...
+		-backend-config="dynamodb_table=uds-ci-state-dynamodb"; \
+	terraform apply -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
+	kubectl get nodes; \
+	terraform destroy -var="ami_id=$${TEST_AMI_ID}" -auto-approve; \
+	snapshot_ids=$$(aws ec2 describe-images --image-ids "$${TEST_AMI_ID}" | jq -r .Images[].BlockDeviceMappings[].Ebs.SnapshotId); \
+	aws ec2 deregister-image --image-id $${TEST_AMI_ID}; \
+	for snapshot in $${snapshot_ids}; do \
+		echo "$${snapshot}"; \
+		aws ec2 delete-snapshot --snapshot-id "$${snapshot}"; \
+	done
```
Comment (mjnagel) [inline Makefile:51]: To improve the CI process this should be 2 separate make make targets. Names can be whatever make sense but the general idea would be:
- tf init + apply make target
- tf destroy + snapshot delete target (maybe even split this into two targets?)
- e2e target would call validate, publish, and those two

The benefits this gives us:
- readability in CI, breaking out each step into its own github section
- ability to _always_ call the destroy pieces, even if the apply fails
- still maintain a single "magic" target for local dev to do the full e2e process

File: test/e2e/rke2/variables.tf [hunk 8]
```diff
@@ -0,0 +1,52 @@
+variable "name" {
+  description = "Name for cluster"
+  type        = string
+  default     = "dubbd-rke2-ci"
```
Comment (mjnagel) [inline test/e2e/rke2/variables.tf:4]:
```suggestion
  default     = "packer-rke2-ci"
```
Missed this one earlier - would be good to name accordingly to help track CI resources.

**Review by mjnagel — COMMENTED** [review]
Soft approval pre-slash command change, will approve once that is modified.

**Review by mjnagel — APPROVED** [review]
:shipit:

**Pull request merged**